### PR TITLE
4.x type cleanup

### PR DESCRIPTION
**Pull request by ADmad**
- Remove deprecated property
- Immutable date time classes are now used by default

### CODE REVIEW

**Comment by ADmad** [comment]
Not sure why only Postgres builds are failing.

**Comment by markstory** [comment]
Maybe there are postgres specific tests?

**Comment by ADmad** [comment]
@markstory I don't think so.

**Comment by markstory** [comment]
Odd, I have postgres installed locally and can try to reproduce.

**Comment by ADmad** [comment]
@markstory Thanks.

**Comment by ADmad** [comment]
` QueryRegressionTest` passes in isolation. The failures seen on travis occur only when running full test suite.

**Comment by ADmad** [comment]
@markstory Tests fixed.

I removed `BaseType::__debugInfo()`. While debugging the test suite failures I found only getting type name pretty useless. I wanted to check whether internally mutable or non mutable class was being used, heence needed to see value of `DateTimeType::$_datetimeInstance` too.

File: tests/TestCase/ORM/Behavior/TimestampBehaviorTest.php [hunk 1]
```diff
@@ -241,6 +240,9 @@ public function testUseImmutable()
         $entity->clean();
         $this->Behavior->handleEvent($event, $entity);
         $this->assertInstanceOf('Cake\I18n\Time', $entity->modified);
+        // Revert back to using immutable class to avoid causing problems in
+        // other test cases when running full test suite.
+        TypeFactory::build('timestamp')->useImmutable();
```
Comment (markstory) [inline tests/TestCase/ORM/Behavior/TimestampBehaviorTest.php]: :clap: good find.

**Comment by markstory** [comment]
Refs #11934

**Pull request merged**